Background dev server for AI coding agents (#16610)
withastro/astro: .changeset/experimental-background-dev.md

---
'astro': major
---

Adds background dev server management for AI coding agents.

When an AI coding agent is detected, `astro dev` now automatically starts the dev server as a detached background process. This prevents the dev server from blocking the agent's terminal and allows it to continue working while the server runs.

A lock file (`.astro/dev.json`) is written when the dev server starts, recording the server's URL, port, and PID. This prevents duplicate servers from being started for the same project.

#### New flag and subcommands

- `astro dev --background` — Start the dev server as a background process (this is what runs automatically when an agent is detected).
- `astro dev stop` — Stop a running background dev server.
- `astro dev status` — Check if a dev server is running and display its URL, PID, and uptime.
- `astro dev logs` — View logs from a background dev server. Use `--follow` (`-f`) to stream new output as it's written.

> These are also available for human use, but are primarily designed for AI coding agents that start and manage dev servers programmatically.

Reviewer: I'm not sure I like the "also available for human use" but this is a nit, feel free to ignore. I would reword it with something like:

```suggestion
These allow you to start and manage dev servers programmatically and were designed with AI coding agents in mind.
```

---
'astro': major
---

Adds background dev server management for AI coding agents.

> When an AI coding agent is detected (via the [`am-i-vibing`](https://github.com/ascorbic/am-i-vibing) library), `astro dev` now automatically starts the dev server as a detached background process. This prevents the dev server from blocking the agent's terminal and allows it to continue working while the server runs.

Author: ```suggestion
When an AI coding agent is detected, `astro dev` now automatically starts the dev server as a detached background process. This prevents the dev server from blocking the agent's terminal and allows it to continue working while the server runs.
```
